core/service/service.go

```diff
@@ -0,0 +1,86 @@
+package service
+
+import "github.com/smartcontractkit/chainlink/core/services/health"
+
+type (
+	// Service represents a long running service inside the
+	// Application.
+	//
+	// Typically a Service will leverage utils.StartStopOnce to implement these
+	// calls in a safe manner.
+	//
+	// Template
+	//
+	// Mockable Foo service with a run loop
+	//  //go:generate mockery --name Foo --output ../../internal/mocks/ --case=underscore
+	//  type (
+	//  	// Expose a public interface so we can mock the service.
+	//  	Foo interface {
+	//  		service.Service
+	//
+	//  		// ...
+	//  	}
+	//
+	//  	foo struct {
+	//  		// ...
+	//
+	//  		stop chan struct{}
+	//  		done chan struct{}
+	//
+	//  		utils.StartStopOnce
+	//  	}
+	//  )
+	//
+	//  var _ Foo = (*foo)(nil)
+	//
+	//  func NewFoo() Foo {
+	//  	f := &foo{
+	//  		// ...
+	//  	}
+	//
+	//  	return f
+	//  }
+	//
+	//  func (f *foo) Start() error {
+	//  	return f.StartOnce("Foo", func() error {
+	//  		go f.run()
+	//
+	//  		return nil
+	//  	})
+	//  }
+	//
+	//  func (f *foo) Close() error {
+	//  	return f.StopOnce("Foo", func() error {
+	//  		// trigger goroutine cleanup
+	//  		close(f.stop)
+	//  		// wait for cleanup to complete
+	//  		<-f.done
+	//  		return nil
+	//  	})
+	//  }
+	//
+	//  func (f *foo) run() {
+	//  	// signal cleanup completion
+	//  	defer close(f.done)
+	//
+	//  	for {
+	//  		select {
+	//  		// ...
+	//  		case <-f.stop:
+	//  			// stop the routine
+	//  			return
+	//  		}
+	//  	}
+	//
+	//  }
```

Comment:
to remove?

Reply:
It's a template for a service: If writing a new service one can just copy the code from godoc and you get start/stop once, proper goroutine cleanup etc. I'll probably also add a sample healthcheck